File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* modifiedList(vector<int>& nums, ListNode* head) {
        sort(nums.begin(), nums.end());
        ListNode* node=head, *prev=NULL;
        while(node){
            if(binary_search(nums.begin(), nums.end(), node->val)){
                if(node==head){
                    head=node->next;
                    node->next=NULL;
                    node=head;
                }
                else{
                    prev->next=node->next;
                    node->next=NULL;
                    node=prev->next;
                }
            }
            else{
                prev=node;
                node=node->next;
            }
        }
        return head;
    }
};

File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {number[]} nums
 * @param {ListNode} head
 * @return {ListNode}
 */

function binary_search(arr, target) {
    let left = 0;
    let right = arr.length - 1;
    while (left <= right) {
        let mid = Math.floor((left + right) / 2);
        if (arr[mid] === target) return true;
        else if (arr[mid] < target) left = mid + 1;
        else right = mid - 1;
    }
    return false;
}
var modifiedList = function(nums, head) {
    nums.sort(function(a,b){
        return a-b;
    })
    let node=head;
    let prev=null;
    while(node){
        if(binary_search(nums, node.val)){
            if(node==head){
                head=node.next;
                node.next=null;
                node=head;
            }
            else{
                prev.next=node.next;
                node.next=null;
                node=prev.next;
            }
        }
        else{
            prev=node;
            node=node.next;
        }
    }
    return head;
};